lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -791,6 +791,21 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
     Value other = convertScalarToDtype(b, loc, operands[1], dtype);
     return b.create<arith::DivFOp>(loc, self, other);
   }
+  if (auto remScalar = dyn_cast<AtenRemainderScalarOp>(op)) {
+    MLIRContext *context = op->getContext();
+    Type newResultType = converter->convertType(remScalar.getType())
+                             .cast<RankedTensorType>()
+                             .getElementType();
+    Type floatDtype = mlir::FloatType::getF64(context);
+    Value self = convertScalarToDtype(b, loc, payloadArgs[0], floatDtype);
+    Value other = convertScalarToDtype(b, loc, operands[1], floatDtype);
```

Comment:
I think it would be better to not cast to f64. When you cast an integer to a float, you run the risk of losing some information. What I would do is cast both `self` and `other` to `resultType`, and then have an `if-else` depending on if the `resultType` is and integer or a float